MyOnboarding/test/MyOnboardingApp.Api.Tests/Controllers/TodoListControllerTests.cs

```diff
@@ -42,72 +57,580 @@ public async Task Get_NoIdSpecified_ReturnsCorrectResponse()
-            Assert.That(itemFromMessage, Is.EqualTo(expectedItem).UsingItemEqualityComparer());
+            Assert.Multiple(async () =>
+            {
+                await _addNewService
+                    .Received(1)
+                    .AddNewItemAsync(itemToAdd);
+                Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.Created));
+                Assert.That(itemFromMessage, Is.EqualTo(expectedItem).UsingItemEqualityComparer());
+            });
+        }
+
+
+        [Test]
+        public async Task Post_EmptyTextSpecifiedInRequestBody_ReturnsBadRequest()
+        {
+            var itemToAdd = new TodoListItem
+            {
+                Text = "",
+                Id = Guid.Empty,
+                CreationTime = DateTime.MinValue,
+                LastUpdateTime = DateTime.MinValue
+            };
+            const string expectedUrl = "expected/Url";
+            _itemUrlLocator
+                .GetListItemUrl(itemToAdd.Id)
+                .Returns(expectedUrl);
+
+            var message = await _controller.GetMessageFromActionAsync(controller => controller.PostAsync(itemToAdd));
+
+            Assert.Multiple(async () =>
+            {
+                await _addNewService
+                    .Received(0)
+                    .AddNewItemAsync(itemToAdd);
+                Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
+            });
+        }
+
+
+        [Test]
+        public async Task Post_InvalidCreationTimeSpecifiedInRequestBody_ReturnsBadRequest()
+        {
+            var itemToAdd = new TodoListItem
+            {
+                Text = "hello",
+                Id = Guid.Empty,
+                CreationTime = new DateTime(2018, 10, 10),
+                LastUpdateTime = DateTime.MinValue
+            };
+            const string expectedUrl = "expected/Url";
+            _itemUrlLocator
+                .GetListItemUrl(itemToAdd.Id)
+                .Returns(expectedUrl);
+
+            var message = await _controller.GetMessageFromActionAsync(controller => controller.PostAsync(itemToAdd));
+
+            Assert.Multiple(async () =>
+            {
+                await _addNewService
+                    .Received(0)
+                    .AddNewItemAsync(itemToAdd);
+                Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
+            });
+        }
+
+
+        [Test]
+        public async Task Post_InvalidItemReturnedToController_ReturnsBadRequest()
+        {
+            var itemToAdd = new TodoListItem
+            {
+                Text = "hello",
+                Id = Guid.Empty,
+                CreationTime = DateTime.MinValue,
+                LastUpdateTime = DateTime.MinValue
+            };
+            var invalidItem = ItemWithErrors.Create(itemToAdd, new List<Error>(new[] { new Error(ErrorCode.DataValidationError, "error", "error.location") }));
```

Comment:
```suggestion
            var invalidItem = ItemWithErrors.Create(itemToAdd, new[] { new Error(ErrorCode.DataValidationError, "error", "error.location") });
```
(based on https://github.com/petrareb/kentico-onboarding-cs/pull/4#discussion_r263424985)

Reply:
There are still some
```csharp
new List<Error> { error }
```
that could be replaced for
```csharp
new [] { error }
```
😉